tests/test_otel.py

```diff
@@ -257,8 +258,40 @@ def test_batches(client, trace_service, batch_count):
     assert len(trace_service.batches) == batch_count
 
     for batch in trace_service.batches:
-        assert get_attr(batch[0].resource, "service.name") == "test_service"
+        assert (
+            get_attr(batch[0].resource, "service.name")
+            == "unknown_service:nginx"
+        )
         assert len(batch[0].scope_spans[0].spans) == batch_size
 
     time.sleep(0.3)  # wait for +1 request to be flushed
     trace_service.batches.clear()
+
+
+@pytest.mark.parametrize(
+    "nginx_config",
+    [
+        {
+            "service_name": 'otel_service_name "test_service";',
+            "res_attrs": 'otel_resource_attr my.name "my name";\n'
+            + 'otel_resource_attr my.service "my service";',
+        }
+    ],
+    indirect=True,
+)
+def test_custom_resource_attributes(client, trace_service):
+    assert client.get("http://127.0.0.1:18080/ok").status_code == 200
+
+    for _ in range(10):
```

Comment:
Let's encapsulate all these details into get_batch() method of TraceService. And "wait and assert" part might deserve its own method, e.g. wait_one_batch() to be used in get_span() as well.

Reply:
Introduced wait_one_batch.